src/tests/functional/shared_test_classes/include/shared_test_classes/base/benchmark.hpp

```diff
@@ -0,0 +1,181 @@
+//
+
+#pragma once
+
+#include <cctype>
+#include <iostream>
+
+#include "layer_test_utils.hpp"
+#include "ov_subgraph.hpp"
+
+namespace LayerTestsDefinitions {
+
+template <typename BaseLayerTest>
+class BenchmarkLayerTest : public BaseLayerTest {
+    static_assert(std::is_base_of<LayerTestsUtils::LayerTestsCommon, BaseLayerTest>::value,
+                  "BaseLayerTest should inherit from LayerTestsUtils::LayerTestsCommon");
+
+public:
+    void Run(const std::initializer_list<std::string>& names,
+             const std::chrono::milliseconds warmupTime = std::chrono::milliseconds(2000),
+             const int numAttempts = 100) {
+        bench_names_ = names;
+        warmup_time_ = warmupTime;
+        num_attempts_ = numAttempts;
+        LayerTestsUtils::LayerTestsCommon::configuration = {{"PERF_COUNT", "YES"}};
+        LayerTestsUtils::LayerTestsCommon::Run();
+    }
+
+    void Run(const std::string& name,
+             const std::chrono::milliseconds warmupTime = std::chrono::milliseconds(2000),
+             const int numAttempts = 100) {
+        if (!name.empty()) {
+            Run({name}, warmupTime, numAttempts);
+        } else {
+            Run({}, warmupTime, numAttempts);
+        }
+    }
+
+    void Validate() override {
+        // NOTE: Validation is ignored because we are interested in benchmarks results.
+        //       In future the validate method could check if new benchmark results are not worse than previous one
+        //       (regression test), and in case of any performance issue report it in PR.
+    }
+
+protected:
+    void Infer() override {
+        std::map<std::string, long long> results_us{};
+        for (const auto& name : bench_names_) {
+            results_us[name] = {};
+        }
+
+        // Warmup
+        auto warm_current = std::chrono::steady_clock::now();
+        const auto warm_end = warm_current + warmup_time_;
+        while (warm_current < warm_end) {
+            LayerTestsUtils::LayerTestsCommon::Infer();
+            warm_current = std::chrono::steady_clock::now();
+        }
+
+        // Benchmark
+        for (int i = 0; i < num_attempts_; ++i) {
+            LayerTestsUtils::LayerTestsCommon::Infer();
+            const auto& perf_results = LayerTestsUtils::LayerTestsCommon::inferRequest.GetPerformanceCounts();
+            for (auto& res : results_us) {
+                const std::string name = res.first;
+                long long& time = res.second;
+                auto found_profile = std::find_if(perf_results.begin(), perf_results.end(),
+                    [&name](const InferenceEngine::InferenceEngineProfileInfo& profile) {
+                        return profile.layer_type == name;
+                    });
+                assert(found_profile != perf_results.end());
+                time += found_profile->second.realTime_uSec;
+            }
+        }
+
+        long long total_us = 0;
+        for (auto& res : results_us) {
```

Comment:
As for me the good option is report that info to somefile
We can use it just local

Reply:
Do you mean to add additional option and print either in console or in file ?

Do you suggest some particular file format ?
Any desired file name ?

Reply:
I meant to report to console and to file simultaneously. 
But it's not required right now (I think that we can reuse the functionality in the CI to report perfomance counters, but I've researched the area and decided not to reuse it according different limitation). I think we can do it later because the main scenario is local debugging

Reply:
How do I know how much performance improvement is it for specific ISA? Do I have to run the test case twice - one with the non-optimized code, and the other with the optimized code, and then get the perf numbers from Console?

Reply:
Guys, please add reporting to the file under the `#ifdef`

Reply:
Which file ? What name of the file do you prefer ? Where to put this file ?

Reply:
`slt_benchmark.txt` in bin dir